libs/langchain/langchain/document_loaders/obsidian.py

```diff
@@ -44,8 +64,14 @@ def _parse_front_matter(self, content: str) -> dict:
         if not match:
             return {}
 
+        self._placeholders: Dict[str, str] = {}
```

Comment:
does this need to be an attribute on the object? can we just pass it in to restore_template_vars and replace_template_var? prefer to avoid mutating state within the DocumentLoader object unless necessary